### ISSUE TITLE
PlayStation analog and rumble support list

### ISSUE THREAD

**Issue by sergiobenrocha2** [libretro/libretro-database#64]
Simplified list of analog and rumble support list. You need to change Input --> Input User "x" Binds --> User "x" Device Type to DualShock as well.

Game                                                                                        | Analog | Rumble
----------------------------------------------------------------------------- | --------- | ---------
Ape Escape                                                                               | Yes      | Yes
Bio Hazard/Resident Evil - Director's Cut - Dual Shock Ver.     | Yes      | Yes
Bio Hazard/Resident Evil 2 - Dual Shock Ver.                           | Yes      | Yes
Bio Hazard/Resident Evil 3 - Nemesis                                       | Yes      | Yes
Breath of Fire IV                                                                        | Yes      | Yes
Chrono Cross                                                                            | Yes      | Yes
Crash Bandicoot 2 - Cortex Strikes Back (USA & Europe) / Crash Bandicoot 2 - Cortex no Gyakushuu! (Japan)          | Yes      | No
Crash Bandicoot 3 - Warped (Europe) / Crash Bandicoot - Warped (USA) / Crash Bandicoot 3 - Buttobi! Sekai Isshuu (Japan)          | Yes      | Yes
Crash Bash (USA & Europe) / Crash Bandicoot Carnival (Japan)         | Yes      | Yes
Crash Bash & Spyro - Year of the Dragon (USA) (Demo)         | Yes      | Yes
CTR - Crash Team Racing (USA & Europe) / Crash Bandicoot Racing (Japan)         | Yes      | Yes
Dance Dance Revolution                                                           | Yes*    | Yes
Dino Crisis                                                                                 | No       | Yes 
Dino Crisis 2                                                                              | Yes     | Yes
Final Fantasy VIII                                                                      | Yes      | Yes
Final Fantasy IX                                                                        | Yes      | Yes
Gran Turismo                                                                            | Yes      | Yes
Gran Turismo 2                                                                        | Yes      | Yes
Haverst Moon                                                                           | Yes*     | No
Jet Moto 3                                                                                 | Yes      | Yes
Metal Gear Solid                                                                       | Yes      | Yes
Metal Slug X                                                                             | No        | Yes**
Mr. Driller                                                                                  | No        | Yes
Need for Speed II (USA & Europe) / Over Drivin' II (Japan)      | Yes      | No
Need for Speed III - Hot Pursuit (USA & Europe) / Over Drivin' III - Hot Pursuit (Japan)         | Yes      | No
Need for Speed - High Stakes (USA) / Need for Speed - Road Challenge (Europe) / Over Drivin' IV (Japan)        | Yes      | Yes
Need for Speed - Porsche Unleashed (USA) / Need for Speed - Porsche 2000 (Europe)       | Yes      | Yes
Need for Speed - V-Rally 2 (USA) / V-Rally - Championship Edition 2 (Europe)       | Yes      | Yes
Parasite Eve                                                                              | Yes      | No
Parasite Eve II                                                                           | Yes      | Yes
Populous - The Beginning                                                         | Yes      | Yes
Porsche Challenge                                                                    | Yes      | No
Quake II                                                                                    | Yes      | Yes
Saltwater Sportfishing                                                               | Yes      | Yes
Spyro 2 - Ripto's Rage! (USA) / Spyro 2 - Gateway to Glimmer (Europe) / Spyro and Sparx - Tondemo Tours (Japan)                                                            | Yes      | Yes
Spyro the Dragon                                                                      | Yes      | Yes
Spyro - Year of the Dragon                                                       | Yes      | Yes
Star Ocean - The Second Story                                                | Yes      | Yes
Tekken 3                                                                                   | No       | Yes
Um Jammer Lammy                                                                  | Yes      | Yes
Vagrant Story                                                                            | Yes      | Yes
WipEout 3                                                                                  | Yes      | Yes

( * ) Acts more like a d-pad
( ** ) The joypad only works if analog sticks are disable, i.e. ANALOG red LED is off in the real console (Core Options --> DualShock analog toggle = ON)

Based on this list: http://ngemu.com/threads/list-of-psx-games-supporting-ds-vibration.111778/

I Tested almost all games (but not all regions), some of the link were wrong, like Populous, Spyro and Tekken 3. I don't know if it's useful to include DualAnalog and DualShock support/tags, since DualAnalog in japan had rumble, and in western didn't.

Still incomplete...

V-Rally - Championship Edition [Platinum Version] seems to be dual shock compatible.

**Comment by sergiobenrocha2** [comment]
tomb raider 3, 4 and 5 have both, confirmed by evdw87

**Comment by hamins123** [comment]
hi do you know how to make a ps1 game dualshock compatible vibration ? if not ..do you someone that can help ? thanks a lotttt

**Comment by mccorkled** [comment]
You can add Tony Hawk 3 & 4 to this list as well. 

I believe Twisted metal 2 & 3 is Dual Shock as well but I cannot get my controllers to work for those games for some reason.

**Comment by simias** [comment]
>( ** ) The joypad only works if analog sticks are disable, i.e. ANALOG red LED is off in the real console (Core Options --> DualShock analog toggle = ON)

This is true of Parasite Eve 1 too: when the ANALOG mode is enabled the D-Pad becomes completely inactive. It's fairly frustrating IMO, I like to navigate the menus with the D-Pad usually, but I guess they were just beginning to figure out how to use analog controls...

**Comment by Rican7** [comment]
I'm leaving this comment/note here for anyone else that finds this issue through search or otherwise, after going down quite a rabbit hole:

 - This list is nice, but old and far from exhaustive
 - As OP mentioned, this is based on a forum thread here: https://www.ngemu.com/threads/list-of-psx-games-supporting-ds-vibration.111778/
 - I found a much more exhaustive, but "old" list here: http://www.rlauncher.com/archive/index.php/t-4408.html

Either way, what was even more confusing to find, was that this very repo actually has internal `.dat` files for supporting this kind of meta:

 - [PlayStation (PS1/PSX) analog support metadata (list)](https://github.com/libretro/libretro-database/blob/v1.9.0/metadat/analog/Sony%20-%20PlayStation.dat)
 - [PlayStation (PS1/PSX) rumble support metadata (list)](https://github.com/libretro/libretro-database/blob/v1.9.0/metadat/rumble/Sony%20-%20PlayStation.dat)

I'm especially mentioning these, because it seems that the presence of these could even mean that this issue could be closed, and we could instead direct people to contribute to those `.dat` files.

Either way, its useful to link to these, as they didn't show up for me through other search means (Google or repo search), so this should surface them a bit more. 🙂

**Comment by mccorkled** [comment]
So does it auto load when it detects the game? I came here because mine was never working like that. How is it working for you?

**Comment by Rican7** [comment]
> So does it auto load when it detects the game?

No, no, it doesn't.

I'm simply saying that we **have the data**, and can improve it going forward.

The cores still need to implement using the data, though.

**Comment by treminaor** [comment]
If this is meant to be a complete list, it's missing quite a few games. Ridge Racer R4 being one of them.

**Comment by mccorkled** [comment]
> 
> 
> If this is meant to be a complete list, it's missing quite a few games. Ridge Racer R4 being one of them.

I actually have a list of the files needed to complete this but never got around to it because I needed some help.

**Comment by gingerbeardman** [comment]
Radikal Bikers (Europe)
- analog YES
- rumble YES

I'll try changing my dat files locally and if it works I'll do a PR.

**Comment by gingerbeardman** [comment]
OK, editing the .dat files and generating updated .rdb files adds support for rumble and analog! 🥳

1. Add your games into `metadat/analog/Sony%20-%20PlayStation.dat` and `metadat/rumble/Sony%20-%20PlayStation.dat`
2. rebuild the .rdb files (`make build`)
3. Copy the udated .rdb file/s into your RetroArch setup  (`retroarch/database/rdb`)
4. Restart RetroArch
   - just to be safe don't use any existing save states for the core you use
   - load saves from memory card

Here are the changes I needed to make to add analog and rumble for Radikal Bikers (Europe): https://github.com/libretro/libretro-database/pull/1312

Notes
- Analog, I could get working before these changes
- Rumble, was unattainable for me until I made the new .rdb file/s

My setup
- RetroArch 1.10.3
- GameForce Chi running EmuElec

> I'm especially mentioning these, because it seems that the presence of these could even mean that this issue could be closed, and we could instead direct people to contribute to those `.dat` files.

I think this is the way forward!

**Comment by photonometric** [comment]
> I'm leaving this comment/note here for anyone else that finds this issue through search or otherwise, after going down quite a rabbit hole:
>
>     * I found a much more exhaustive, but "old" list here: http://www.rlauncher.com/archive/index.php/t-4408.html

Thanks a lot for leaving those links! As you suspected, this is one of the most useful pages that comes up in a google search. Very useful for general game historian/researchers, who aren't necessarily interested in advanced emulation topics. 

The RocketLauncher Forums seem to have upgrade to a newer forum software, and thread format. Here's the updated link:


[RLauncher Forums - **Sony PlayStation: Controller Compatibility Matrix and List of Special Settings**](
https://www.rlauncher.com/forum/index.php?threads/sony-playstation-controller-compatibility-matrix-and-list-of-special-settings.4408/) (2017-02)

**Comment by OctopusButtons** [comment]
> leaving this comment/note here for anyone else that finds this issue through search or otherwise, after going down quite a rabbit hole

Thank you for leaving all these carrots, said a visiting rabbit who stuck its head down the rabbit hole.

**Comment by epiChrisCH** [comment]
Delta Force Urban Warfare uses the modern fps both sticks setup.
